src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/DeleteAccountButton.tsx

```diff
@@ -21,29 +22,57 @@ export const DeleteAccountButton = (
     onDeleteAccount: typeof onDeleteAccount;
   },
 ) => {
+  const l10n = useL10n();
   const recordTelemetry = useTelemetry();
   const [isSubmitting, setIsSubmitting] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
   return (
-    <Button
-      {...props}
-      isLoading={isSubmitting}
-      isDisabled={isSubmitting}
-      onPress={() => {
-        recordTelemetry("ctaButton", "click", {
-          button_id: "confirm_delete_account",
-        });
-        setIsSubmitting(true);
-        // It's currently unclear if and how we should mock our server action:
-        /* c8 ignore next 8 */
-        void props
-          .onDeleteAccount()
-          .then(() => {
-            void signOut({ callbackUrl: "/" });
-          })
-          .catch(() => {
-            setIsSubmitting(false);
+    <>
+      <Button
+        {...props}
+        isLoading={isSubmitting}
+        isDisabled={isSubmitting}
+        onPress={() => {
+          recordTelemetry("ctaButton", "click", {
+            button_id: "confirm_delete_account",
           });
-      }}
-    />
+          setIsSubmitting(true);
+          setErrorMessage(null);
+          // It's currently unclear if and how we should mock our server action:
+          /* c8 ignore next 13 */
+          void props
+            .onDeleteAccount()
+            .then((result) => {
+              if (result && typeof result === "object" && "success" in result) {
+                if (result.success === false) {
+                  setIsSubmitting(false);
+                  setErrorMessage(
+                    result.errorMessage ??
+                      l10n.getString("settings-delete-account-error-generic"),
+                  );
+                  return;
+                }
+              }
+              void signOut({ callbackUrl: "/" });
+            })
+            .catch(() => {
+              setIsSubmitting(false);
+              setErrorMessage(
+                l10n.getString("settings-delete-account-error-generic"),
+              );
+            });
+        }}
+      />
+      {errorMessage && (
+        <p
+          role="status"
+          style={{
+            marginTop: "0.5rem",
```

Comment:
Probably doesn't matter for such a small and rarely visible surface, but we tend to avoid `margin`s in CSS because they hurt the composability of individual elements. Instead, we try to be consistent in having layout "flow down", e.g. having a flexbox parent container with a `gap`.

Again, probably not a biggie for this specific code, but good to know.